applications/GeoMechanicsApplication/custom_elements/interface_element.cpp

```diff
@@ -303,6 +304,15 @@ std::vector<Vector> InterfaceElement::CalculateTractionsAtIntegrationPoints(cons
     return result;
 }
 
+void InterfaceElement::ApplyRotationToBMatrix(Matrix& rBMatrix, const Matrix& rRotationMatrix) const
+{
+    const auto dim = GetGeometry().WorkingSpaceDimension();
+    for (int i = 0; i + dim <= rBMatrix.size2(); i += dim) {
+        auto sub_matrix = subrange(rBMatrix, 0, rBMatrix.size1(), i, i + dim);
+        sub_matrix.assign(Matrix{prod(sub_matrix, trans(rRotationMatrix))});
```

Comment:
You used here trans(rRotationMatrix) replacing of R^-1. This is valid as long as the rotation matrix is orthogonal. However, I suppose that the rotation matrix is built based on this concept, otherwise a check is needed.